compiler/src/iree/compiler/Codegen/Common/test/vector_layout_analysis.mlir

```diff
@@ -145,3 +145,46 @@ builtin.module attributes { transform.with_named_sequence } {
     transform.yield 
   }
 }
+
+// -----
+
+#layout = #iree_vector_ext.layout<<[VECTORY], [16]>, <[BATCHY, VECTORX], [2, 8]>>
+
+// Propagate and enforce through scf.for
+builtin.module attributes { transform.with_named_sequence } {
+  func.func @scffor(%arr: memref<16x16xf16>, %arr2: memref<16xf16>, %a: vector<16xf16>, %b: vector<16xf16>) -> vector<16xf16> {
+    %c0 = arith.constant 0 : index
+    %c1 = arith.constant 1 : index
+    %c1024 = arith.constant 1024 : index
+    %cst_0 = arith.constant 0.0 : f16
+    %cst0_1 = arith.constant dense<0.0> : vector<16xf16>
+    // expected-remark @above {{layout of result #0 is #iree_vector_ext.layout<<[ VECTORY], [16]>>}}
+
+    %out = scf.for %iv = %c0 to %c1024 step %c1 iter_args(%arg1 = %cst0_1) -> (vector<16xf16>) {
+      // expected-remark @above {{layout of result #0 is #iree_vector_ext.layout<<[ VECTORY], [16]>>}}
+      %root = vector.transfer_read %arr[%c0, %c0], %cst_0 {in_bounds = [true, true], "__vector_layout_test_anchor_result_0" = #layout} : memref<16x16xf16>, vector<16x16xf16>
+      // expected-remark @above {{layout of result #0 is #iree_vector_ext.layout<<[ VECTORY], [16]>, <[ BATCHY,  VECTORX], [2, 8]>>}}
```

Comment:
tbh not immediately obvious to me what this is fixing from the LIT test, seems like the anchored layout is the transfer_read, was the layout of the transfer read not being propagated before? Would be nice to have an check-not.

Reply:
scf.for wasn't tested before which is why I added this test. This test also checks if the analysis went till fixed-point, which it wasn't before because of the bug.